pkg/provisioner/ironic/testbmc/testbmc.go

```diff
@@ -0,0 +1,83 @@
+package testbmc
+
+import (
+	"net/url"
+
+	"github.com/metal3-io/baremetal-operator/pkg/bmc"
+)
+
+func init() {
+	bmc.RegisterFactory("test", newTestBMCAccessDetails, []string{})
+	bmc.RegisterFactory("test-needs-mac", newTestBMCAccessDetails, []string{})
+}
```

Comment:
Would it be safer to register these explicitly in tests rather than it happening as a side-effect here?

Reply:
Probably. On the other hand, all of the rest of the registry stuff expects to be invoked from an init() function like this, the registry is global state without a way to reset right now, and I'm not sure what happens if the same name is registered more than once.

I would be happy to consider making the registry more accepting of dynamic use in a separate PR.